HTMLSerializer.js

```diff
@@ -353,6 +374,80 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Load all external fonts, and add an entry to |this.html| at index
+   * |this.fontPlaceHolderIndex|.
+   *
+   * @param {Document} doc The Document being serialized.
+   */
+  loadFonts(doc) {
+    this.fontPlaceHolderIndex = this.html.length;
+    this.html.push(''); // Entry where the font style tag will go.
+    for (var i = 0, styleSheet; styleSheet = doc.styleSheets[i]; i++) {
+      if (styleSheet.cssRules) {
+        for (var j = 0, rule; rule = styleSheet.cssRules[j]; j++) {
+          this.processCSSFonts(doc.defaultView, styleSheet.href, rule.cssText);
+        }
+      } else {
+        this.crossOriginStyleSheets.push(styleSheet.href);
+      }
+    }
+  }
+
+  /**
+   * Takes a string representing CSS and parses it to find any fonts that are
+   * declared.  If any fonts are declared, it processes them so that they
+   * can be used in the serialized document and adds them to |this.fontCSS|.
+   *
+   * @param {Window} win The Window of the Document being serialized.
+   * @param {string} href The url at which the CSS stylesheet is located.
+   * @param {string} css The CSS text.
+   */
+  processCSSFonts(win, href, css) {
+    var serializer = this;
+    var fonts = css.match(/@font-face *?{.*?}/g);
+    if (fonts) {
+      var nestingDepth = this.windowDepth(win);
+      var escapedQuote = this.escapedCharacter('"', nestingDepth);
+      for (var i = 0; i < fonts.length; i++) {
+        // Convert url specified in font to fully qualified url.
+        var font = fonts[i].replace(/url\("(.*?)"\)/g, function(match, url) {
+          url = serializer.fullyQualifiedFontURL(href, url);
+          return 'url("' + url + '")';
+        }).
+        replace(/"/g, escapedQuote);
+        this.fontCSS.push(font);
+      }
+    }
+  }
+
+  /**
+   * Computes the fully qualified url at which a font can be loaded.
+   *
+   * @param {string} href The url at which the CSS stylesheet containing the
+   *     font is located.
+   * @param {string} url The url listed in the font declaration.
+   */
+  fullyQualifiedFontURL(href, url) {
```

Comment:
The method as is does not work for all image urls.  I could try to extend it, but there are a lot of edge cases I'm running into.  For example, and image url could be a data url, or about:blank, neither of which are handled by this method.  Do you suggest I try to extend it?